sdks/go/pkg/beam/io/rtrackers/offsetrange/offsetrange_test.go

```diff
@@ -299,3 +301,225 @@ func TestTracker_TrySplit(t *testing.T) {
+// before claiming anything.
+func TestGrowableTracker_SplitBeforeStart(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	estimator.SetEstimateRangeEnd(10)
+	p, r, _ := tracker.TrySplit(0)
+
+	expected := Restriction{0, 0}
+	if p.(Restriction) != expected {
+		t.Errorf("wrong primaries after TrySplit(0), got: %v, want: %v", p.(Restriction), expected)
+	}
+	if tracker.GetRestriction().(Restriction) != expected {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0), got: %v, want: %v", tracker.GetRestriction().(Restriction), expected)
+	}
+	res := Restriction{0, math.MaxInt64}
+	if res != r.(Restriction) {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expected)
+	}
+}
+
+// TestGrowableTracker_CheckpointJustStarted tests TryClaim and TrySplit
+// for GrowableTracker.
+func TestGrowableTracker_CheckpointJustStarted(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	if !tracker.TryClaim(int64(5)) {
+		t.Fatal("tracker.TryClaim(int64(5)) should've claimed.")
+	}
+	estimator.SetEstimateRangeEnd(0)
+	p, r, _ := tracker.TrySplit(0)
+	if tracker.IsDone() {
+		t.Fatal("tracker not done yet, , but IsDone() returns true")
+	}
+
+	expPr := Restriction{0, 6}
+	if p.(Restriction) != expPr {
+		t.Errorf("wrong primaries after TrySplit(0), got: %v, want: %v", p.(Restriction), expPr)
+	}
+	if tracker.GetRestriction().(Restriction) != expPr {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0), got: %v, want: %v", tracker.GetRestriction().(Restriction), expPr)
+	}
+	expRes := Restriction{6, math.MaxInt64}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expRes)
+	}
+
+	tracker, err = NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	if !tracker.TryClaim(int64(5)) {
+		t.Fatal("tracker.TryClaim(int64(5)) should've claimed.")
+	}
+	estimator.SetEstimateRangeEnd(20)
+	p, r, _ = tracker.TrySplit(0)
+	if tracker.IsDone() {
+		t.Fatal("tracker not done yet, , but IsDone() returns true")
+	}
+	if p.(Restriction) != expPr {
+		t.Errorf("wrong primaries after TrySplit(0), got: %v, want: %v", p.(Restriction), expPr)
+	}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expRes)
+	}
+}
+
+// TestGrowableTracker_Split tests TrySplit method for GrowableTracker
+// in regards to returned primary and residual.
+func TestGrowableTracker_Split(t *testing.T) {
```

Comment:
Consider generalizing this into a table-driven test to check a number of restriction sizes and split points.

Reply:
Done